pkg/ruler/manager.go

```diff
@@ -141,33 +137,50 @@ func (r *DefaultMultiTenantManager) syncRulesToManager(ctx context.Context, user
 		return
```

Comment:
Above we call `r.mapper.MapRules()`. It's not thread safe. Currently `syncRulesToManager()` should be never called concurrently, but we haven't documented it, so at least we should update function comments to clarify it.

Reply:
Added a comment about this not being thread safe for the same user. I'm not sure if it's completely not safe due to the underlying `MkdirAll`  call.

Reply:
I think the race would be in the creation of new files + deletion of old ones.